.github/workflows/workflow.yml

```diff
@@ -7,91 +7,160 @@ jobs:
-      run: |
-        make --directory ./docs generate
-        
-    - name: Build documentation website
-      run: |
-        make --directory ./docs html
-
-    - uses: actions/upload-artifact@v1
-      with:
-        name: docs
-        path: docs/build/html
+      - name: "1. Clone repository"
+        uses: actions/checkout@v2
 
-  deploy-docs:
+      - name: "2. Set up Python 3.6"
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.6
+
+      - name: "3. Update pip and install pandoc"
+        run: |
+          sudo apt-get install -y pandoc
+          python -m pip install --upgrade pip
+
+      - name: "4. Attach cache for pip"
+        uses: actions/cache@v1
+        id: cache
+        with:
+          path: ~/.cache/pip
+          key: docs-${{ runner.os }}-pip-${{ hashFiles('requirements/main.txt', 'requirements/release.txt') }}
+          restore-keys: |
+            docs-${{ runner.os }}-pip-
+
+      - name: "5. Install main and release dependencies"
+        run: |
+          pip install -q .
+          pip install -q -r requirements/release.txt
+
+      - name: "6. Copy examples"
+        run: |
+          make --directory ./docs copy-examples
+
+      - name: "7. Generate API documentation"
+        run: |
+          make --directory ./docs generate
+
+      - name: "8. Build documentation"
+        run: |
+          make --directory ./docs html
+
+      - name: "9. Store documentation artifact"
+        uses: actions/upload-artifact@v1
+        with:
+          name: docs
+          path: docs/build/html
+
+  release:
     needs: [tests, docs]
     runs-on: ubuntu-latest
     if: contains(github.ref, 'master')
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/download-artifact@v1
+      - name: "1. Clone repository"
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+
+      - name: "2. Set up Python 3.6"
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.6
+
+      - name: "3. Update pip and install pandoc"
+        run: |
+          sudo apt-get install -y pandoc
+          python -m pip install --upgrade pip
+
+      - name: "4. Install release dependencies"
+        run: |
+          pip install -q -r requirements/release.txt
+
+      - name: "5. Determine next version"
+        uses: endrjuskr/github-tag-action@master
+        id: bump_dry
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          INITIAL_VERSION: 0.0.2
+          DRY_RUN: true
+
+      - name: "6. Bump version"
+        run: |
+          bump2version --current-version ${{ steps.bump_dry.outputs.tag }} --new-version ${{ steps.bump_dry.outputs.new_tag }} patch setup.py
+
+      - name: "7. Commit new version"
+        uses: stefanzweifel/git-auto-commit-action@v4.1.6
+        with:
+          commit_message: Release ${{ steps.bump_dry.outputs.new_tag }}
+          branch: master
+          commit_user_name: Unit8 Bot
```

Comment:
This is not ideal, because we create new commit on master, but we need to do it somewhere

Reply:
Also such use does not exist

Reply:
can it be created in the PR branch instead?

Reply:
do you mean manually on develop? or we branch from develop and merge it to master?

Reply:
hmm what about the trigger to bump version (so steps up to including 8) being on pull_request to master? and on push to master, the create tag and release draft is created?

to avoid touching the develop branch, guess we could just create a release branch from develop and a PR of that to master (instead of directly develop to master)

Reply:
Discussed offline, we stick with current approach